etcdserver/etcdhttp/client.go

```diff
@@ -92,7 +92,7 @@ func NewClientHandler(server *etcdserver.EtcdServer) http.Handler {
 	mux := http.NewServeMux()
 	mux.HandleFunc("/", http.NotFound)
 	mux.Handle(healthPath, healthHandler(server))
-	mux.HandleFunc(versionPath, serveVersion)
+	mux.HandleFunc(versionPath, versionHandler(server.Cluster(), serveVersion))
```

Comment:
probably not. we still want to test the http handler and we also pass in a closure so that we do not need to construct a handler struct that only contains a cluster field.